### ISSUE TITLE
Create or Update Bibliography

### ISSUE THREAD

**Issue by jeirf12** [JabRef/jabref#12262]
The point is that when you click on the update bibliography button, you should be able to choose with the cursor where to start creating the section. Also, have the same format of the document, for example IEEE. Also, it should not create a different section than the one currently being worked on.

**Comment by koppor** [comment]
@jeirf12 Please provide more context. Maybe a screenshot?

I think, you are talking about the LibreOffice integration of JabRef?

**Comment by subhramit** [comment]
@jeirf12 If I understand your issue correctly:
1. You desire that Bibliography is generated/updated at the place where cursor is present.
2. You desire that JabRef automatically detects the style of the existing bibliography
3. You desire that there are no two bibliography sections created for the same document

I would like to know which version of JabRef you are using - it is asked in the issue format so that we have more context. There have been major changes in JabRef's OO/LO integration since the last release.

My thoughts:
1. It will be very convoluted to generate bibliography at the position of the cursor. Reason being in case of the auto-generate bibliography feature, which generates and updates the bibliography on the run as you keep citing items in the rest of the document (which are inserted at the cursor position), there will be conflict (whether to insert a citation or generate a bibliography at that position?).
By default, the Bibliography section is generated at the end of text to avoid such conflicts between citations and bibliography. 
There could be, however, a provision made in the case of the auto-generate feature being turned off - such that when "generate/update bibliography" is pressed, it generates it at the position of the cursor - but I don't see the utility of this - as the bibliography section is always supposed to be after the text at the end of the document anyway? 
In case you really require the section to be somewhere else, you can always cite your items (with auto generate bibliography turned off), then generate the bibliography once you are done - and then move it to your desired place. Just a matter of four clicks.

2. You have the provision to choose the style you want by defining it in JStyle files (in released versions at the time of this comment). If you download the unreleased version, you can choose pre-formatted citation (and bibliography) styles as well - such as IEEE, by using CSL Styles in the "Select Style" dialog of the Open/LibreOffice panel. Reverse-parsing the currently present style in the document and selecting that is a near impossible task without AI, and a very difficult task with AI.

3. In the released versions, two separate bibliography sections are never created, except if you manually create your own bibliography section (or use any other software to do that) and then use JabRef to generate it again. The reason being JabRef uses it's own unique mechanism to mark the section in the text and then identify, edit/modify or refresh it. In the unreleased version, two bibliography sections are created only when you mix up the style families you are using in the document - that is, if you use both a JStyle for some references, and CSl Styles for other. Within each family, there will always be one single bibliography section created pertaining to it.

CC: @ThiloteE , @koppor

**Comment by subhramit** [comment]
Closing as what was within scope has been addressed.

**Comment by koppor** [comment]
Maybe we could add some documentation to the LO plugin?

**Comment by subhramit** [comment]
> Maybe we could add some documentation to the LO plugin?

Yes. A lot of CSL documentation is also pending since the beginning (have been a little lazy on that front). Will try to add some documentation with @ThiloteE together before 6.0 is released. He has already made a project/issue at the user documentation repository, if I remember correctly.
